### DISCUSSION TITLE
Dont Resetting after the form

### DISCUSSION THREAD

**Discussion by fatihdurmaz** [filamentphp/filament#12613] [General]
How do I prevent any field from resetting after the form is submitted?

**Comment by UNM0** [comment]
Can u provide a code snippet ?? Of you form?!!

**Reply by fatihdurmaz** [reply]
I don't want the select box to be reset.

**Comment by jechazelle** [comment]
You must not use:

```
$this->form->fill()
```

**Reply by fatihdurmaz** [reply]
class SubjectResource extends Resource
{
    protected static ?string $model = Subject::class;

    protected static ?string $navigationIcon = 'heroicon-o-document-text';

    protected static ?string $modelLabel = 'Konu';

    protected static ?string $navigationGroup  = 'GYUD';

    protected static ?string $pluralModelLabel = 'Konular';

    protected static ?string $slug = 'konular';

    public static function infolist(Infolist $infolist): Infolist
    {
        return $infolist
            ->schema([
                TextEntry::make('subject_name')
                    ->label('Konu Adı')
                    ->columnSpanFull(),
            ]);
    }

    public static function form(Form $form): Form
    {
        return $form
            ->schema([

                Section::make()
                    ->schema([

                        Textarea::make('subject_name')
                            ->label('Konu Adı')
                            ->autosize()
                            ->unique(ignoreRecord: true)
                            ->required(),

                        Select::make('subject_type')
                            ->options(SubjectType::class)
                            ->prefixIcon('heroicon-o-arrows-up-down')
                            ->label('Konu Türü')
                            ->required(),

                        Toggle::make('status')
                            ->label('Durum')
                            ->default(true)
                            ->onColor('success')
                            ->offColor('danger')
                    ])
                    ->columns(1)
            ]);
    }

    public static function table(Table $table): Table
    {
        return $table
            ->columns([

                TextColumn::make('subject_name')
                    ->label('Konu Adı')
                    ->searchable()
                    ->wrap()
                    ->icon('heroicon-o-document-text')
                    ->sortable(),

                TextColumn::make('subject_type')
                    ->badge()
                    ->label('Konu Türü'),

                IconColumn::make('status')
                    ->label('Durum')
                    ->boolean(),

            ])
            ->striped()
            ->filters([
                SelectFilter::make('subject_type')
                    ->label('Konu Türü')
                    ->options(SubjectType::class),
            ])
            ->actions([
                Tables\Actions\EditAction::make(),
            ])
            ->bulkActions([
                Tables\Actions\BulkActionGroup::make([
                    Tables\Actions\DeleteBulkAction::make(),
                ]),
            ]);
    }

    public static function getRelations(): array
    {
        return [
            QuestionsRelationManager::class,
            SummariesRelationManager::class,
        ];
    }

    public static function getPages(): array
    {
        return [
            'index'     => Pages\ListSubjects::route('/'),
            'create'    => Pages\CreateSubject::route('/create'),
            'edit'      => Pages\EditSubject::route('/{record}/edit'),
        ];
    }

    public static function getNavigationBadge(): ?string
    {
        return static::getModel()::count();
    }
}
Where
